api/src/main/java/eu/darkbot/api/game/other/LocationInfo.java

```diff
@@ -29,15 +31,21 @@ public interface LocationInfo extends Location {
      * Calculates future destination of entity in time(ms), if entity isn't moving returns copy of location.
      */
     default Location destinationInTime(long time) {
-        Location copy = copy();
-        double speed = getSpeed();
+        return destinationInTime(copy(), time);
+    }
 
+    /**
+     * Adds the distance to {@code target} which entity can travel in given time(ms)<br>
+     * Similar to {@link #destinationInTime(long)} but does not create the copy
+     */
+    default Location destinationInTime(@NotNull Location target, long time) {
```

Comment:
this is using an out-param which is generally a bad pattern, but if you're going to allow it at least explicitly call it that, and return void:
```suggestion
    default void destinationInTime(@NotNull Location result, long time) {
```
(and explain in the docs that result will be explicitly mutated to add the movement)

On top of that, i don't get how this is supposed to be useful, because it potentially opens up to you using one arbitrary location and adding to it the hero's (or another entity's) direction/angle movement, which seems like a bug. Wouldn't you always want to call this with your own current location, and so, needing the copy?